src/hotspot/share/gc/parallel/parallelScavengeHeap.cpp

```diff
@@ -337,7 +337,9 @@ HeapWord* ParallelScavengeHeap::mem_allocate_work(size_t size,
 }
 
 void ParallelScavengeHeap::do_full_collection(bool clear_all_soft_refs) {
-  PSParallelCompact::invoke(clear_all_soft_refs);
+  // No need for max-compaction in this context.
+  const bool should_do_max_compaction = false;
+  PSParallelCompact::invoke(clear_all_soft_refs, should_do_max_compaction);
```

Comment:
Perhaps this might do too, without the need for an additional variable?
```suggestion
  PSParallelCompact::invoke(clear_all_soft_refs, false /* should_do_max_compaction */);
```

Reply:
True; I chose this way mostly for consistency -- `satisfy_failed_allocation` uses local var instead of comment.